pandas/tseries/base.py

```diff
@@ -811,6 +812,23 @@ def summary(self, name=None):
         result = result.replace("'", "")
         return result
 
+    def _append_same_dtype(self, to_concat, name):
+        """
+        Concatenate to_concat which has the same class
+        """
+        attribs = self._get_attributes_dict()
```

Comment:
this is fine, but has bothered me that we have to be Index congnizant of which attributes need propgation (e.g. ``freq`` for DTI is optional, but required for ``Period`` so have to care, similarly with ``tz`` which is needed for DTI but not for ``Period``) and not to mention that neither matter for TDI.

just a comment, not really much can/should do about this. Good that you are moving the special case logic in a single place. Maybe add a comment / assertion as needed here to show guarantees (if that makes sense)